### PR TITLE
Deployment to AWS

### PR DESCRIPTION
**Pull request by matthewhall79**
This branch adds support to deploy the app to AWS. I suggest that reviewers start by reading the updated `docs/DEVELOPERS.md` and then the new `docs/AWS-account.md`. (Note that the steps documented in the latter won't need to be repeated unless we ever wish to provision an entirely new AWS account in which to deploy the app, but they give an overview of how AWS deployment is achieved.)

Note that the contents of the `amplify` directory is mostly autogenerated by the AWS Amplify CLI tool.

There are a few caveats / jobs still to be done:

- Coach assignment is the only function that current works in production instances (AWS instances) of the app. Further work is required to allow adding and editing of coaches (these functions still work in local development instances).
- The Python backend still implements the coach database as a JSON file; this needs to be migrated to AWS Dynamo.
- There are a couple of short blocks of code in `Assign.js` related to making HTTP requests to the backend that should be commonized for use in `Add.js` and `Edit.js`; these are marked with `TODO` comments.

### CODE REVIEW

File: draw-arch-diagram [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        backend  = Lambda("Backend app")
+        db       = Dynamodb("Coach DB")
+
+    user >> frontend
+
+    js >> frontend
+    python >> backend
+
+    frontend >> [auth, backend]
+    backend >> [auth, db]
```
Comment (LewisGaul) [inline draw-arch-diagram:33]: Python as a DSL, crazy stuff ;)

**Comment by LewisGaul** [comment]
Pretty cool, good work putting this together and appreciate the inclusion of the writeup and diagram.

To give this a full review I'd try setting it all up from scratch following your instructions, but that doesn't sound massively appealing - let me know if you think there'd be high value in this, otherwise I'm happy to approve based on a brief look through the changes.

**Comment by matthewhall79** [comment]
> To give this a full review I'd try setting it all up from scratch following your instructions, but that doesn't sound massively appealing - let me know if you think there'd be high value in this, otherwise I'm happy to approve based on a brief look through the changes.

No, I don't think it would be worthwhile. Having set everything up in my existing AWS account, I then did everything again from scratch in a new account created specifically for the Crisis app. I wrote the instructions during that process. There's a good chance we'll never have to go through the steps again, but I thought they would be a useful reference for understanding how the account is set up.

At some point soon someone will develop some changes to the frontend and want to test out deploying them using the Amplify CLI, at which point we'll need to flesh out and record the instructions for doing this.

File: amplify/cli.json [hunk 2]
```diff
@@ -1,3 +1,19 @@
 {
```
Comment (LewisGaul) [inline amplify/cli.json:1]: Intentionally made executable? Wouldn't have expected this to be intentional for a json file!

**Review by LewisGaul — APPROVED** [review]

File: amplify/cli.json (hunk 2 above)
Comment (matthewhall79) [reply amplify/cli.json:1]: Seems wrong to me too, but it's generated by the Amplify CLI so I'm inclined to leave it as it is.

**Pull request merged**